homeassistant/components/recorder.py

```diff
@@ -302,7 +304,7 @@ def block_till_done(self):
 
     def _setup_connection(self):
         """Ensure database is ready to fly."""
-        db_path = self.hass.config.path(DB_FILE)
+        db_path = ':memory:' if self.pytest else self.hass.config.path(DB_FILE)
```

Comment:
Yes, that would be a better solution. The pytest documentation said making the code "test-aware" was typically not the right way to do things but didn't mention what the correct way would be.

On April 8, 2016 5:28:36 PM EDT, Paulus Schoutsen <notifications@github.com> wrote:
>> @@ -302,7 +304,7 @@ def block_till_done(self):
>>  
>>      def _setup_connection(self):
>>          """Ensure database is ready to fly."""
>> -        db_path = self.hass.config.path(DB_FILE)
>> +        db_path = ':memory:' if self.pytest else
>self.hass.config.path(DB_FILE)
>
>We could mock out config.path instead of adding a pytest option 
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/balloob/home-assistant/pull/1752/files/db242b473f7d9d02c44271a7b837998c083fb7ea#r59091896